src/EventRegistry.h

```diff
@@ -22,18 +22,55 @@ enum class EventGroupKind {
     Module,
 };
 
+class EnumVal;
 class EventGroup;
 class EventHandler;
 class EventHandlerPtr;
 class RE_Matcher;
+class RecordVal;
+class Type;
```

Comment:
Just as an observation, we seem to declare these same type aliases in multiple places into the same namespace. This seems error-prone and we should probably just declare them once and `#include` that definition (possibly hard since not all headers might have been designed for that).

Reply:
You're thinking about something like broker's [fwd.hh](https://github.com/zeek/broker/blob/master/libbroker/broker/fwd.hh) ? Maybe something like `ValFwd.h` ?